roles/ceph-dashboard/tasks/configure_dashboard.yml

```diff
@@ -150,7 +150,9 @@
   changed_when: false
 
 - name: set grafana api password
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-password {{ grafana_admin_password }}"
+  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-password -i -"
```

Comment:
You need to also add the `--interactive` container option there I think (after `container_exec_cmd`)